### ISSUE TITLE
Lint error when needlessly supressing a lint

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#4636]
Rubocop (a Ruby linter) can suppress lints via a magic comment in a similar way that you can `#[allow(clippy::...)]`. Rubocop _also_ has a lint that triggers if one of these comments appears and is not suppressing a lint error.

One common thing I've run into is to have many `#[allow(clippy::module_name_repetitions)]` and then rename the module in a refactoring, which makes these properties not required and they will remain as noise.

Is such a lint feasible?

**Comment by flip1995** [comment]
Sadly this can hardly be done in Clippy and needs to be implemented in rustc.

> Is such a lint feasible?

But yes, of course this is very feasible! There even is an [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2383-lint-reasons.md#expect-lint-attribute) about this feature. Also see the tracking issue: https://github.com/rust-lang/rust/issues/54503.

TL;DR: The plan is to introduce an `expect` attribute into the compiler, that does just that.
